### PR TITLE
increase thieving beacon range to 2

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
make it slightly outrange fulton beacons random offset

## Why / Balance
greentext getting trolled because i fultoned an objective but it was just outside the steal area so i had to ftl to my hiding spot, nudge it, then evac launched before i could get back

thieves are explicitly intended to use fultons to steal stuff so this just makes it less awful

## Technical details
changed a number

## Requirements
- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- tweak: Increased the thieving beacon's range to 2 tiles.

### CODE REVIEW

**Comment by TheShuEd** [comment]
as far as i remember, there is no interaction check, and now you can count the beacon behind the wall of the storage facility, and the radius will hit the nuke

**Comment by mierdapovareshka** [comment]
бля чё это за говно ебаное

**Comment by deltanedas** [comment]
no longer able to steal things through walls :trollface:

**Comment by slarticodefast** [comment]
Doesn't InRangeUnobstructed cause problems if you have two steal objectives next to each other, where one blocks the other? For example in the screenshot above if both the ChemMaster and ChemDispenser were both steal targets. Also did you check if objects inside chests in range still count?

**Comment by deltanedas** [comment]
because of how it checks it recursively it would search the crates contents, even if the crate obstructed it

File: Content.Server/Objectives/Systems/StealConditionSystem.cs [hunk 1]
```diff
@@ -101,15 +103,18 @@ private float GetProgress(MindComponent mind, StealConditionComponent condition)
...
             {
                 if (!area.Owners.Contains(mind.Owner))
                     continue;
 
                 var nearestEnt = _lookup.GetEntitiesInRange(uid, area.Range);
                 foreach (var ent in nearestEnt)
                 {
+                    if (!_interaction.InRangeUnobstructed((uid, xform), ent, range: area.Range))
+                        continue;
+
```
Comment (metalgearsloth) [inline Content.Server/Objectives/Systems/StealConditionSystem.cs:117]: 1. Use GetEntitiesInRange component version, it should already be returning contained entities with the query.
2. Use hashset cache.

File: Content.Server/Objectives/Systems/StealConditionSystem.cs [hunk 2]
```diff
@@ -101,15 +105,19 @@ private float GetProgress(MindComponent mind, StealConditionComponent condition)
         //check stealAreas
         if (condition.CheckStealAreas)
         {
-            var areasQuery = AllEntityQuery<StealAreaComponent>();
-            while (areasQuery.MoveNext(out var uid, out var area))
+            var areasQuery = AllEntityQuery<TransformComponent, StealAreaComponent>();
+            while (areasQuery.MoveNext(out var uid, out var xform, out var area))
```
Comment (metalgearsloth) [inline Content.Server/Objectives/Systems/StealConditionSystem.cs:109]: Remember that this iterates literally every entity ingame.

Comment (deltanedas) [reply Content.Server/Objectives/Systems/StealConditionSystem.cs:109]: isnt the most common comp supposed to be first arg

Comment (metalgearsloth) [reply Content.Server/Objectives/Systems/StealConditionSystem.cs:109]: Without Arch it just gets the dictionary of the first component and then trycomps every single entity in that list.

Comment (deltanedas) [reply Content.Server/Objectives/Systems/StealConditionSystem.cs:109]: bruh

**Comment by TheShuEd** [comment]
Webedit fail

**Pull request merged**